iceoryx_hoofs/include/iceoryx_hoofs/internal/relocatable_pointer/pointer_repository.inl

```diff
@@ -138,6 +114,27 @@ inline id_t PointerRepository<id_t, ptr_t, CAPACITY>::searchId(ptr_t ptr) const
     /// by setting id to RAW_POINTER_BEHAVIOUR_ID
     return RAW_POINTER_BEHAVIOUR_ID;
 }
+template <typename id_t, typename ptr_t, uint64_t CAPACITY>
+inline bool PointerRepository<id_t, ptr_t, CAPACITY>::addPointerIfIndexIsFree(const id_t id,
```

Comment:
OK, checked (with @FerdinandSpitzschnueffler ). This refactoring is shorter, equivalent and introduces no overhead (good refactoring).